src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/GetDownloadResultUtility.cs

```diff
@@ -21,14 +23,18 @@ public static class GetDownloadResultUtility
            PackageIdentity identity,
            Uri uri,
            ISettings settings,
+           SourceCacheContext cacheContext,
            ILogger logger,
            CancellationToken token)
         {
             // Uri is not null, so the package exists in the source
             // Now, check if it is in the global packages folder, before, getting the package stream
 
-            // TODO: This code should respect no_cache settings and not write or read packages from the global packages folder
-            var packageFromGlobalPackages = GlobalPackagesFolderUtility.GetPackage(identity, settings);
+            DownloadResourceResult packageFromGlobalPackages = null;
+
+            //TODO: NuGet/Home#1406 This code should respect -NoCache option and not read packages from the global packages folder
+            //Note cacheContext.NoCache indicates that packages are not written to the global packages folder
+            packageFromGlobalPackages = GlobalPackagesFolderUtility.GetPackage(identity, settings);
```

Comment:
If there is something in the cache, -DirectDownload won't have effect (at least that's what the code looks like and my own testing locally indicates). Is this expected?